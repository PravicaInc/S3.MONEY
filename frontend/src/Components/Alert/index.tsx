import { FC, ReactNode } from 'react';
import { twMerge } from 'tailwind-merge';

import AlertIcon from '@/../public/images/alert_icon.svg?jsx';
import BackgroundModalDecorativeIcon from '@/../public/images/background_modal_decorative_left.svg?jsx';

import { Button } from '@/Components/Form/Button';
import { Modal, ModalProps } from '@/Components/Modal';

export interface AlertProps extends ModalProps {
  onOkClick: () => void;
  okButtonContent?: ReactNode;
  header?: ReactNode;
  description?: ReactNode;
}

export const Alert: FC<AlertProps> = ({
  onClose,
  onOkClick,
  header,
  description,
  okButtonContent = 'OK',
  className,
  ...props
}) => (
  <Modal
    onClose={onClose}
    className={twMerge('relative p-6 w-[400px]', className)}
    {...props}
  >
    <div className="absolute top-0 left-0 z-[-1]">
      <BackgroundModalDecorativeIcon />

      <div className="absolute top-6 left-6 bg-mistyRose w-12 h-12 flex items-center justify-center rounded-full">
        <AlertIcon />
      </div>
    </div>
    <p className="text-primary text-lg font-semibold mt-16">
      {header}
    </p>
    <p className="mt-1 text-secondary">
      {description}
    </p>
    <Button
      className="mt-8 w-full h-11"
      onClick={onOkClick}
    >
      {okButtonContent}
    </Button>
  </Modal>
);
