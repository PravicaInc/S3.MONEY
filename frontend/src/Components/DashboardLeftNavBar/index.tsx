'use client';

import { FC, HTMLAttributes, useMemo } from 'react';
import Link from 'next/link';
import { usePathname, useSearchParams } from 'next/navigation';
import { twMerge } from 'tailwind-merge';

import LogoIcon from '@/../public/images/logo.svg?jsx';
import OperationsLinkIcon from '@/../public/images/operations_link_icon.svg?jsx';

import { PAGES_URLS } from '@/utils/const';

export const DashboardLeftNavBar: FC<HTMLAttributes<HTMLDivElement>> = ({ className, ...props }) => {
  const searchParams = useSearchParams();
  const pathname = usePathname();

  const links = useMemo(
    () => [
      {
        groupName: 'Main Menu',
        groupLinks: [
          {
            href: {
              pathname: PAGES_URLS.dashboardOperations,
              query: Object.fromEntries(searchParams.entries()),
            },
            icon: <OperationsLinkIcon />,
            text: 'Operations',
            isActive: pathname === PAGES_URLS.dashboardOperations,
          },
        ],
      },
    ],
    [searchParams, pathname]
  );

  return (
    <div
      className={twMerge(
        'h-screen w-[270px] border-r border-borderPrimary bg-white',
        className
      )}
      {...props}
    >
      <Link
        href={PAGES_URLS.home}
        className="ml-6 my-5 block"
      >
        <LogoIcon />
      </Link>
      {
        links.map(({ groupName, groupLinks }) => (
          <div key={groupName} className="pt-5 px-4">
            <p className="text-xs font-semibold text-hitGrey pl-3 pb-3">
              {groupName}
            </p>
            <div className="flex flex-col gap-1">
              {groupLinks.map(({ text, href, icon, isActive }) => (
                <Link
                  key={text}
                  href={href}
                  className={twMerge(
                    'flex items-center gap-[10px] py-2 px-3 border rounded-md border-transparent text-secondary',
                    isActive && 'border-borderPrimary bg-[#F8F9FB] text-primary'
                  )}
                >
                  {icon}
                  {text}
                </Link>
              ))}
            </div>
          </div>
        ))
      }
    </div>
  );
};
