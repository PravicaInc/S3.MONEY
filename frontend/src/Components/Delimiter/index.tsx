import { FC, HTMLAttributes } from 'react';
import { twMerge } from 'tailwind-merge';

export const Delimiter: FC<HTMLAttributes<HTMLDivElement>> = ({ className, ...props }) => (
  <div
    className={twMerge(
      'bg-borderPrimary h-[1px]',
      className
    )}
    {...props}
  />
);
