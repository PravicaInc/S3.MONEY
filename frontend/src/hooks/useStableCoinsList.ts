import { useQuery } from '@tanstack/react-query';
import { AxiosResponse } from 'axios';

import { ApiManager } from '@/services/api';

interface PackageApiResponse {
  deploy_addresses: {
    packageId: string,
    token_policy: string,
    token_supply: string,
    treasury_cap: string,
    token_policy_cap: string,
    pauser: string,
  },
  deploy_date: string,
  deploy_status: 'created' | 'published',
  package_name: string,
  ticker: string,
  ticker_name: string,
  txid: string,
  icon_url: string;
  package_zip: string,
  package_roles: {
    burn: string,
    freeze: string,
    cashIn: string,
    pause: string,
    cashOut: string
  },
  address_roles: string[],
}

export interface StableCoin extends Omit<PackageApiResponse, 'ticker_name' | 'icon_url'> {
  txid: string;
  ticker: string;
  name: string;
  icon?: string;
}

export interface StableCoinsListResponse {
  accountAddress: string;
  coins: StableCoin[];
}

const packagesListApi = new ApiManager(
  process.env.NEXT_PUBLIC_API_DOMAIN as string,
  '/packages'
);

export interface packagesListApiGetByKeyResponse {
  status: 'ok',
  packages: PackageApiResponse[],
}

const getStableCoinsList = async (accountAddress: string): Promise<StableCoinsListResponse> => {
  const { packages } = await packagesListApi.getByKey(accountAddress, { summary: 1 })
    .then((response: AxiosResponse<packagesListApiGetByKeyResponse>) => response.data)
    .then(data => ({
      ...data,
      packages: data.packages
        .filter(({ deploy_status }) => deploy_status === 'published')
        .sort(({ deploy_date: a }, { deploy_date: b }) => (new Date(b)).getTime() - (new Date(a)).getTime()),
    }));

  return {
    accountAddress,
    coins: packages
      .filter(({ deploy_status }) => deploy_status === 'published')
      .sort(({ deploy_date: a }, { deploy_date: b }) => (new Date(b)).getTime() - (new Date(a)).getTime())
      .map(({ txid, ticker_name, ticker, icon_url, ...otherProps }) => ({
        txid,
        ticker,
        name: ticker_name,
        icon: icon_url,
        ...otherProps,
      })),
  };
};

export const useStableCoinsList = (accountAddress: string | undefined) => {
  const query = useQuery<StableCoinsListResponse>({
    queryKey: ['stableCoinsList', accountAddress],
    queryFn: () => accountAddress
      ? getStableCoinsList(accountAddress)
      : Promise.resolve({} as StableCoinsListResponse),
  });

  return {
    ...query,
  };
};
