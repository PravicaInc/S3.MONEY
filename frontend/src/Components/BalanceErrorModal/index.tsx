import { FC, useCallback, useState } from 'react';
import { useSuiClientContext } from '@mysten/dapp-kit';

import AlertIcon from '@/../public/images/alert_icon.svg?jsx';
import BackgroundModalDecorativeIcon from '@/../public/images/background_modal_decorative_left.svg?jsx';

import { Button, BUTTON_VIEWS } from '@/Components/Form/Button';
import { Modal, ModalProps } from '@/Components/Modal';

import { useRequestSuiTokens } from '@/hooks/useRequestSuiTokens';

export interface BalanceErrorModalProps extends ModalProps {}

export const BalanceErrorModal: FC<BalanceErrorModalProps> = ({
  onClose,
  ...props
}) => {
  const suiClientContext = useSuiClientContext();
  const requestSuiTokens = useRequestSuiTokens();

  const [requestSuiTokensSuccess, setRequestSuiTokensSuccess] = useState<boolean>(false);
  const [requestSuiTokensError, setRequestSuiTokensError] = useState<boolean>(false);

  const requestToken = useCallback(
    async () => {
      const faucetRequestStatus = await requestSuiTokens.mutateAsync();

      setRequestSuiTokensSuccess(faucetRequestStatus === 'SUCCEEDED');
      setRequestSuiTokensError(faucetRequestStatus === 'DISCARDED');
    },
    [requestSuiTokens]
  );

  return (
    <Modal
      onClose={onClose}
      className="relative p-6 w-[480px]"
      {...props}
    >
      <div className="absolute top-0 left-0 z-[-1]">
        <BackgroundModalDecorativeIcon />

        <div className="absolute top-6 left-6 bg-mistyRose w-12 h-12 flex items-center justify-center rounded-full">
          <AlertIcon />
        </div>
      </div>
      <p className="text-primary text-lg font-semibold mt-16">
        Your balance is too low
      </p>
      <p className="mt-1 text-secondary">
        Unfortunately, your balance is too small to create a new StableCoin.
        {
          requestSuiTokensError && (
            <>
              <br />
              <span className="text-error">
                Something went wrong, and we were unable to complete your request. Please try again.
              </span>
            </>
          )
        }
      </p>
      <div className="grid grid-cols-4 items-center justify-center gap-4 mt-8">
        <Button
          className="h-[56px] col-span-1"
          view={BUTTON_VIEWS.secondary}
          onClick={onClose}
        >
          Close
        </Button>
        {
          ['testnet', 'devnet'].includes(suiClientContext.network) && (
            <Button
              className="h-[56px] col-span-3"
              onClick={requestToken}
              disabled={requestSuiTokens.isPending || requestSuiTokens.isSuccess}
              isLoading={requestSuiTokens.isPending}
            >
              {
                requestSuiTokensSuccess
                  ? (
                    <>
                      {
                        suiClientContext.network === 'testnet'
                          ? 1
                          : 10
                      }
                      {' '}
                      SUI received
                    </>
                  )
                  : (
                    <>
                      Request
                      <span className="capitalize px-1">
                        {suiClientContext.network}
                      </span>
                      SUI Tokens
                    </>
                  )
              }
            </Button>
          )
        }
      </div>
    </Modal>
  );
};
