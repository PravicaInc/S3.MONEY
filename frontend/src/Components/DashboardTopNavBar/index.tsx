'use client';

import { FC, HTMLAttributes, useMemo } from 'react';
import { useAutoConnectWallet, useCurrentAccount } from '@mysten/dapp-kit';
import { twMerge } from 'tailwind-merge';

import { Loader } from '@/Components/Loader';
import { LogoutButton } from '@/Components/LogoutButton';
import { SelectStableCoinDropdown } from '@/Components/SelectStableCoinDropdown';

import { useShortAccountAddress } from '@/hooks/useShortAccountAddress';

export const DashboardTopNavBar: FC<HTMLAttributes<HTMLDivElement>> = ({ className, ...props }) => {
  const autoConnectionStatus = useAutoConnectWallet();
  const account = useCurrentAccount();
  const shortAccountAddress = useShortAccountAddress();

  const isLoading = useMemo(
    () => autoConnectionStatus === 'idle',
    [autoConnectionStatus]
  );
  const isRedirecting = useMemo(
    () => autoConnectionStatus === 'attempted' && !account?.address,
    [autoConnectionStatus, account?.address]
  );

  return (
    <div
      className={twMerge(
        'flex items-center justify-between p-6 h-24 border-b border-borderPrimary bg-white',
        className
      )}
      {...props}
    >
      <SelectStableCoinDropdown />
      {
        (isLoading || isRedirecting) && (
          <Loader className="h-8" />
        )
      }
      {
        shortAccountAddress && (
          <LogoutButton />
        )
      }
    </div>
  );
};
