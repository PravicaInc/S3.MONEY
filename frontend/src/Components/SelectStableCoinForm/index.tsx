'use client';

import React, { ChangeEvent, FC, HTMLAttributes, useCallback, useMemo, useState } from 'react';
import { useAutoConnectWallet, useCurrentAccount } from '@mysten/dapp-kit';
import Link from 'next/link';
import { twMerge } from 'tailwind-merge';

import SearchIcon from '@/../public/images/search.svg?jsx';

import { Delimiter } from '@/Components/Delimiter';
import { Button } from '@/Components/Form/Button';
import { SimpleInput } from '@/Components/Form/Input';
import { Loader } from '@/Components/Loader';

import { PAGES_URLS } from '@/utils/const';

import { StableCoin, useStableCoinsList } from '@/hooks/useStableCoinsList';

import { StableCoinItem } from './components/StableCoinItem';

export interface SelectStableCoinFormProps extends HTMLAttributes<HTMLDivElement> {}

export const SelectStableCoinForm: FC<SelectStableCoinFormProps> = ({ className, ...props }) => {
  const autoConnectionStatus = useAutoConnectWallet();
  const account = useCurrentAccount();
  const {
    data,
    isLoading: isStableCoinsListLoading,
    isFetching: isStableCoinsListFetching,
  } = useStableCoinsList(account?.address);

  const [searchValue, setSearchValue] = useState<string>('');
  const [selectedStableCoin, setSelectedStableCoin] = useState<StableCoin | null>();

  const { coins: stableCoins = [] } = data || {};

  const filteredStableCoins = useMemo(
    () => stableCoins.filter(coin => stableCoinMatchesSearch(coin, searchValue)),
    [stableCoins, searchValue]
  );
  const isLoading = useMemo(
    () => autoConnectionStatus === 'idle',
    [autoConnectionStatus]
  );
  const isRedirecting = useMemo(
    () => autoConnectionStatus === 'attempted' && !account?.address,
    [autoConnectionStatus, account?.address]
  );

  const changeSearchValue = useCallback(
    (event: ChangeEvent<HTMLInputElement>) => {
      const newValue = event.target.value;

      setSearchValue(newValue);

      if (selectedStableCoin && !stableCoinMatchesSearch(selectedStableCoin, newValue)) {
        setSelectedStableCoin(null);
      }
    },
    [selectedStableCoin]
  );

  return (
    <div
      className={twMerge('bg-white shadow-stableCoinForm rounded-xl', className)}
      {...props}
    >
      <div className="px-6 py-5 flex items-center justify-between border-b border-borderPrimary">
        <span className="text-lg font-semibold text-primary">
          Select Stablecoin
        </span>
        {
          isLoading || isRedirecting
            ? (
              <Loader className="h-5" />
            )
            : (
              <Link href={PAGES_URLS.createStableCoin} className="rounded-xl">
                <Button className="text-sm font-semibold h-[37px] w-[153px]">
                  + New Stablecoin
                </Button>
              </Link>
            )
        }
      </div>
      {
        isLoading || isRedirecting || isStableCoinsListLoading || isStableCoinsListFetching
          ? (
            <div className="flex h-40 items-center justify-center">
              <Loader className="h-10" />
            </div>
          )
          : (
            <div className="p-6 space-y-6">
              <SimpleInput
                className="w-full"
                value={searchValue}
                onChange={changeSearchValue}
                placeholder="Search..."
                icon={<SearchIcon />}
              />
              <Delimiter />
              <div className="space-y-3 max-h-[410px] overflow-auto">
                {
                  filteredStableCoins.map(coin => (
                    <StableCoinItem
                      key={coin.id}
                      stableCoinItem={coin}
                      onClick={() => setSelectedStableCoin(coin)}
                      isSelected={selectedStableCoin?.id === coin.id}
                    />
                  ))
                }
              </div>
              <Button
                className="w-full h-[37px] text-sm font-semibold"
                disabled={!selectedStableCoin}
              >
                Manage Stablecoin
              </Button>
            </div>
          )
      }
    </div>
  );

  function stableCoinMatchesSearch(stableCoin: StableCoin, search: string) {
    return stableCoin.name.toLowerCase().includes(search.toLowerCase())
      || stableCoin.tokenName.toLowerCase().includes(search.toLowerCase());
  }
};
