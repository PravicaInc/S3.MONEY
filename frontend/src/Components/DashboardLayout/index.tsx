import { FC, ReactNode } from 'react';
import { twMerge } from 'tailwind-merge';

import { DashboardLeftNavBar } from '@/Components/DashboardLeftNavBar';
import { DashboardTopNavBar } from '@/Components/DashboardTopNavBar';
import { Footer } from '@/Components/Footer';

export interface DashboardLayoutProps {
  children?: ReactNode;
  className?: string;
  contentClassName?: string;
}

export const DashboardLayout: FC<DashboardLayoutProps> = ({ children, className, contentClassName }) => (
  <div className={twMerge('flex min-h-screen bg-pageBgPrimary', className)}>
    <DashboardLeftNavBar className="shrink-0" />
    <div className={twMerge('w-screen h-screen flex flex-col', contentClassName)}>
      <DashboardTopNavBar className="shrink-0" />
      <div className="overflow-auto h-full flex flex-col justify-between">
        <div className="h-full">
          {children}
        </div>
        <Footer
          className="pb-7 pl-9 pr-[85px]"
        />
      </div>
    </div>
  </div>
);

