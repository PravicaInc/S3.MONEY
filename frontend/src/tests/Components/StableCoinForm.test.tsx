import { useAutoConnectWallet, useCurrentAccount } from '@mysten/dapp-kit';
import { act, cleanup, fireEvent, getQueriesForElement, RenderResult } from '@testing-library/react';

import '@testing-library/jest-dom';

import { SelectStableCoinForm } from '@/Components/SelectStableCoinForm';

import { useStableCoinsList } from '@/hooks/useStableCoinsList';

import {
  createMockStableCoinList,
  TEST_STABLE_COIN1,
  TEST_STABLE_COIN2,
} from '@/tests/utils/create_mock_stable_coin_list';
import { createMockAccount } from '@/tests/utils/create_mock_wallet_account';
import { renderWithProviders } from '@/tests/utils/render_with_providers';

const renderSelectStableCoinFormWithProviders = () => renderWithProviders(<SelectStableCoinForm />);

let renderResult: RenderResult;

describe('SelectStableCoinForm:', () => {
  afterAll(cleanup);

  describe('SelectStableCoinForm with loading state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'idle');

      renderResult = renderSelectStableCoinFormWithProviders();
    });

    it('Show loaders', () => {
      expect(renderResult.queryAllByTestId('loader').length).toBe(2);
    });
  });

  describe('SelectStableCoinForm with disconnected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');

      renderResult = renderSelectStableCoinFormWithProviders();
    });

    it('Show loaders', () => {
      expect(renderResult.queryAllByTestId('loader').length).toBe(2);
    });
  });

  describe('SelectStableCoinForm with connected state:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');
      jest.mocked(useCurrentAccount).mockImplementation(createMockAccount);

      renderResult = renderSelectStableCoinFormWithProviders();
    });

    it('Don`t show loaders instead of the StableCoin creation link', () => {
      expect(renderResult.queryByText(/New Stablecoin/)).toBeVisible();
    });

    it('Show StableCoin list loader', () => {
      expect(renderResult.queryByTestId('loader')).toBeVisible();
    });
  });

  describe('SelectStableCoinForm with StableCoin list:', () => {
    beforeEach(() => {
      if (renderResult?.unmount) {
        renderResult.unmount();
      }

      jest.mocked(useAutoConnectWallet).mockImplementation(() => 'attempted');
      jest.mocked(useCurrentAccount).mockImplementation(createMockAccount);
      jest.mocked(useStableCoinsList).mockImplementation(
        createMockStableCoinList()
      );

      renderResult = renderSelectStableCoinFormWithProviders();
    });

    it('Show StableCoin list', () => {
      expect(renderResult.queryByTestId('loader')).toBeNull();
      expect(renderResult.queryByText(TEST_STABLE_COIN1.name)).toBeVisible();
      expect(renderResult.queryByText(TEST_STABLE_COIN2.name)).toBeVisible();
    });

    it('Show filtered StableCoin list', () => {
      const searchInput = renderResult.getByPlaceholderText(/Search/);

      fireEvent.change(searchInput, { target: { value: TEST_STABLE_COIN2.name } });

      expect(renderResult.queryByText(TEST_STABLE_COIN1.name)).toBeNull();
      expect(renderResult.queryByText(TEST_STABLE_COIN2.name)).toBeVisible();

      fireEvent.change(searchInput, { target: { value: '' } });

      expect(renderResult.queryByText(TEST_STABLE_COIN1.name)).toBeVisible();
      expect(renderResult.queryByText(TEST_STABLE_COIN2.name)).toBeVisible();
    });

    it('Select StableCoin', () => {
      const buttonStableCoin1 = renderResult.getByRole('button', {
        name: new RegExp(TEST_STABLE_COIN1.name),
      });
      const buttonStableCoin2 = renderResult.getByRole('button', {
        name: new RegExp(TEST_STABLE_COIN2.name),
      });

      act(() => {
        buttonStableCoin1.click();
      });

      expect(
        (getQueriesForElement(buttonStableCoin1).queryByRole('checkbox') as HTMLInputElement)?.checked
      ).toBeTruthy();

      act(() => {
        buttonStableCoin2.click();
      });

      expect(
        (getQueriesForElement(buttonStableCoin1).queryByRole('checkbox') as HTMLInputElement)?.checked
      ).toBeFalsy();
      expect(
        (getQueriesForElement(buttonStableCoin2).queryByRole('checkbox') as HTMLInputElement)?.checked
      ).toBeTruthy();
    });

    it('Unselect StableCoin if it has been filtered', () => {
      const buttonStableCoin1 = renderResult.getByRole('button', {
        name: new RegExp(TEST_STABLE_COIN1.name),
      });
      const searchInput = renderResult.getByPlaceholderText(/Search/);

      act(() => {
        buttonStableCoin1.click();
      });

      fireEvent.change(searchInput, { target: { value: TEST_STABLE_COIN2.name } });
      fireEvent.change(searchInput, { target: { value: '' } });

      expect(
        (getQueriesForElement(renderResult.getByRole('button', {
          name: new RegExp(TEST_STABLE_COIN1.name),
        })).queryByRole('checkbox') as HTMLInputElement)?.checked
      ).toBeFalsy();
    });

    it('Disabled and enabled manage stablecoin button', () => {
      const manageStablecoinButton = renderResult.getByText('Manage Stablecoin') as HTMLButtonElement;
      const buttonStableCoin1 = renderResult.getByRole('button', {
        name: new RegExp(TEST_STABLE_COIN1.name),
      });
      const searchInput = renderResult.getByPlaceholderText(/Search/);

      expect(manageStablecoinButton?.disabled).toBeTruthy();

      act(() => {
        buttonStableCoin1.click();
      });

      expect(manageStablecoinButton?.disabled).toBeFalsy();

      fireEvent.change(searchInput, { target: { value: TEST_STABLE_COIN1.name } });

      expect(manageStablecoinButton?.disabled).toBeFalsy();

      fireEvent.change(searchInput, { target: { value: TEST_STABLE_COIN2.name } });

      expect(manageStablecoinButton?.disabled).toBeTruthy();

      fireEvent.change(searchInput, { target: { value: '' } });

      expect(manageStablecoinButton?.disabled).toBeTruthy();
    });

    it('Snapshot', () => {
      expect(renderResult.container).toMatchSnapshot();
    });
  });
});
