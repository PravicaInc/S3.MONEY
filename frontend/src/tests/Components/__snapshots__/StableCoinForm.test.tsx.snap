// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectStableCoinForm: SelectStableCoinForm with StableCoin list: Snapshot 1`] = `
<div>
  <style>
    [data-dapp-kit] {--dapp-kit-blurs-modalOverlay:blur(0);--dapp-kit-backgroundColors-primaryButton:#F6F7F9;--dapp-kit-backgroundColors-primaryButtonHover:#F0F2F5;--dapp-kit-backgroundColors-outlineButtonHover:#F4F4F5;--dapp-kit-backgroundColors-modalOverlay:rgba(24 36 53 / 20%);--dapp-kit-backgroundColors-modalPrimary:white;--dapp-kit-backgroundColors-modalSecondary:#F7F8F8;--dapp-kit-backgroundColors-iconButton:transparent;--dapp-kit-backgroundColors-iconButtonHover:#F0F1F2;--dapp-kit-backgroundColors-dropdownMenu:#FFFFFF;--dapp-kit-backgroundColors-dropdownMenuSeparator:#F3F6F8;--dapp-kit-backgroundColors-walletItemSelected:white;--dapp-kit-backgroundColors-walletItemHover:#3C424226;--dapp-kit-borderColors-outlineButton:#E4E4E7;--dapp-kit-colors-primaryButton:#373737;--dapp-kit-colors-outlineButton:#373737;--dapp-kit-colors-iconButton:#000000;--dapp-kit-colors-body:#182435;--dapp-kit-colors-bodyMuted:#767A81;--dapp-kit-colors-bodyDanger:#FF794B;--dapp-kit-radii-small:6px;--dapp-kit-radii-medium:8px;--dapp-kit-radii-large:12px;--dapp-kit-radii-xlarge:16px;--dapp-kit-shadows-primaryButton:0px 4px 12px rgba(0, 0, 0, 0.1);--dapp-kit-shadows-walletItemSelected:0px 2px 6px rgba(0, 0, 0, 0.05);--dapp-kit-fontWeights-normal:400;--dapp-kit-fontWeights-medium:500;--dapp-kit-fontWeights-bold:600;--dapp-kit-fontSizes-small:14px;--dapp-kit-fontSizes-medium:16px;--dapp-kit-fontSizes-large:18px;--dapp-kit-fontSizes-xlarge:20px;--dapp-kit-typography-fontFamily:ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";--dapp-kit-typography-fontStyle:normal;--dapp-kit-typography-lineHeight:1.3;--dapp-kit-typography-letterSpacing:1;}
  </style>
  <div
    class="bg-white shadow-stableCoinForm rounded-xl"
  >
    <div
      class="px-6 py-5 flex items-center justify-between border-b border-borderPrimary"
    >
      <span
        class="text-lg font-semibold text-primary"
      >
        Select Stablecoin
      </span>
      <a
        class="rounded-xl"
        href="/create-stablecoin"
      >
        <button
          class="text-white rounded-xl shadow-button border-actionSecondary border-solid border flex items-center justify-center bg-buttonBgMain relative overflow-hidden after:content-[''] after:bg-buttonBgAfter after:w-full after:h-full after:absolute hover:bg-buttonBgAfter hover:text-actionPrimary transition disabled:bg-slate-300 disabled:bg-none disabled:border-slate-400 disabled:text-white text-sm font-semibold h-[37px] w-[153px]"
        >
          + New Stablecoin
        </button>
      </a>
    </div>
    <div
      class="p-6 space-y-6"
    >
      <div
        class="relative"
      >
        <input
          class="text-primary placeholder:text-hitGrey p-3 rounded-xl border border-borderPrimary focus:border-actionPrimary outline-none pl-11 w-full"
          placeholder="Search..."
          type="text"
          value=""
        />
        <svg
          class="absolute top-1/2 left-3 -translate-y-1/2"
        />
      </div>
      <div
        class="bg-borderPrimary h-[1px]"
      />
      <div
        class="space-y-3 max-h-[410px] overflow-auto"
      >
        <button
          class="border rounded-[10px] px-4 py-3 flex items-center justify-between w-full text-left border-borderPrimary"
        >
          <div
            class="flex gap-4 items-center"
          >
            <div
              class="w-8 h-8 rounded-full bg-seashell bg-no-repeat bg-center bg-cover"
              style="background-image: url(undefined);"
            />
            <div>
              <p
                class="font-semibold text-primary"
              >
                PRV
              </p>
              <p
                class="text-sm text-grayText mt-[2px]"
              >
                Pravica Token
              </p>
            </div>
          </div>
          <label
            class="relative transition cursor-pointer"
          >
            <input
              class="
  invisible absolute inset-1/2
  [&+div]:checked:border-actionPrimary [&+div]:checked:bg-actionPrimary
  [&+div>svg]:checked:block
"
              readonly=""
              type="checkbox"
            />
            <div
              class="w-5 h-5 border border-borderPrimary flex items-center justify-center rounded-md transition"
            >
              <svg
                class="hidden duration-1000"
              />
            </div>
          </label>
        </button>
        <button
          class="border rounded-[10px] px-4 py-3 flex items-center justify-between w-full text-left border-borderPrimary"
        >
          <div
            class="flex gap-4 items-center"
          >
            <div
              class="w-8 h-8 rounded-full bg-seashell bg-no-repeat bg-center bg-cover"
              style="background-image: url(undefined);"
            />
            <div>
              <p
                class="font-semibold text-primary"
              >
                SSS
              </p>
              <p
                class="text-sm text-grayText mt-[2px]"
              >
                S3 Money Token
              </p>
            </div>
          </div>
          <label
            class="relative transition cursor-pointer"
          >
            <input
              class="
  invisible absolute inset-1/2
  [&+div]:checked:border-actionPrimary [&+div]:checked:bg-actionPrimary
  [&+div>svg]:checked:block
"
              readonly=""
              type="checkbox"
            />
            <div
              class="w-5 h-5 border border-borderPrimary flex items-center justify-center rounded-md transition"
            >
              <svg
                class="hidden duration-1000"
              />
            </div>
          </label>
        </button>
      </div>
      <button
        class="text-white rounded-xl shadow-button border-actionSecondary border-solid border flex items-center justify-center bg-buttonBgMain relative overflow-hidden after:content-[''] after:bg-buttonBgAfter after:w-full after:h-full after:absolute hover:bg-buttonBgAfter hover:text-actionPrimary transition disabled:bg-slate-300 disabled:bg-none disabled:border-slate-400 disabled:text-white w-full h-[37px] text-sm font-semibold"
        disabled=""
      >
        Manage Stablecoin
      </button>
    </div>
  </div>
</div>
`;
